src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -673,6 +679,129 @@ public void testOrderOfCheckExecution() throws Exception {
         }
     }
 
+    @Test
+    public void testSkipFileOnJavaParseExceptionTrue() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "true");
+        config.addProperty("javaParseExceptionSeverity", "ignore");
+        config.addChild(createModuleConfig(ConstantNameCheck.class));
+
+        final File[] files = {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+            new File(getPath("InputTreeWalkerProperFileExtension.java")),
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+        };
+
+        final Checker checker = createChecker(config);
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                Collections.singletonList(
+                        "10:27: " + getCheckMessage(ConstantNameCheck.class,
+                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$")));
+        verify(checker, files, expectedViolation);
+    }
+
+    @Test
+    public void testSkipFileOnJavaParseExceptionFalse() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "false");
+        config.addChild(createModuleConfig(ConstantNameCheck.class));
+
+        final String[] files = {
+            getNonCompilablePath("InputTreeWalkerSkipParsingException.java"),
+            getPath("InputTreeWalkerProperFileExtension.java"),
+            getNonCompilablePath("InputTreeWalkerSkipParsingException.java"),
+        };
+
+        try {
+            execute(config, files);
+            assertWithMessage("Exception is expected").fail();
+        }
+        catch (CheckstyleException exception) {
+            assertWithMessage("Error message is unexpected")
+                    .that(exception.getMessage())
+                    .contains("Exception was thrown while processing ");
+        }
```

Comment:
Please use https://github.com/checkstyle/checkstyle/blob/bfb3e5f5e416211bc95799426edba4581b442d0d/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java#L393 in all cases like this.

Reply:
Done